atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1338,23 +1341,38 @@ public String toString() {
                 for (Entry<Cell, Value> entry : cells) {
                     Value value = entry.getValue();
                     String tableName = new String(entry.getKey().getRowName());
-                    if (currentlyExistingTables.contains(tableName)) {
-                        byte[] contents;
-                        if (value == null) {
-                            contents = AtlasDbConstants.EMPTY_TABLE_METADATA;
-                        } else {
-                            contents = value.getContents();
-                        }
-                        tableToMetadataContents.put(tableName, contents);
+
+                    byte[] contents;
+                    if (value == null) {
+                        contents = AtlasDbConstants.EMPTY_TABLE_METADATA;
                     } else {
-                        log.info("Non-existing table {}: {}", tableName, value);
+                        contents = value.getContents();
                     }
+                    tableToMetadataContents.put(tableName, contents);
                 }
             }
         } finally {
             range.close();
         }
-        return tableToMetadataContents;
+
+        final Set<String> tablesInDatabase = getAllTableNames();
```

Comment:
do you also want to check that the table actually exists in getMetadataForTable (singular)?